src/mame/misc/gameace.cpp

```diff
@@ -212,50 +455,81 @@ ROM_START( hotbodya ) // sprites and sound section ROMs match the above, tilemap
 	ROM_LOAD( "pal4.1d",  0x600, 0x157, NO_DUMP ) // PALCE20V8H-25PC/4
 ROM_END
 
-
-void gameace_state::init_hotbody()
+void gameace_state::decode_cpu()
 {
-	// TODO:  enough for data but opcodes appear to have different scrambling
-	uint8_t *rom = memregion("maincpu")->base();
+	uint8_t* rom = memregion("maincpu")->base();
 	std::vector<uint8_t> buffer(0x40000);
 	memcpy(&buffer[0], rom, 0x40000);
 
-	for (int i = 0x00000; i < 0x40000; i++) // TODO: simplify this
+	for (int i = 0x00000; i < 0x40000; i++)
 	{
+		uint8_t swap_code[0x10] = { 0x0a, 0x07, 0x08, 0x05,   0x06, 0x03, 0x04, 0x01,   0x02, 0x0f, 0x00, 0x0d,   0x0e, 0x0b, 0x0c, 0x09 };
+		uint8_t swap_data[0x10] = { 0x03, 0x02, 0x01, 0x00,   0x05, 0x04, 0x07, 0x06,   0x0d, 0x0c, 0x0f, 0x0e,   0x09, 0x08, 0x0b, 0x0a };
```

Comment:
If these don’t change as you walk the ROM, you should make them `constexpr`, and you should also move them outside the loop.